cirq-core/cirq/vis/heatmap.py

```diff
@@ -299,6 +302,30 @@ def plot(
             ax = cast(plt.Axes, ax)
         original_config = copy.deepcopy(self._config)
         self.update_config(**kwargs)
+
+        highlighted_qubits = kwargs.get("highlighted_qubits")
```

Comment:
If `highlighted_qubits=None`, which is the default, `set(highlighted_qubits)` will be inappropriate, so we will need to do `set([highlighted_qubits])` in this case, which means the default value of `highlighted_qubits` must be `[None]` instead of `None`. If so, the condition `if highlighted_qubits is not None` must also be changed to become a more complex condition. These might be too many changes for a single change and the behavior doesn't change when using set(). However, if you think we need to still perform these changes, please let me know and I can surely make the change.

About the latter, `if highlighted_qubits` raises

```
ValueError: The truth value of an array with more than one element is ambiguous. Use a.any() or a.all()
```

if `highlighted_qubits` is a numpy array but doesn't yield any error if it's a `list`. Whereas `if highlighted_qubits is not None` seems more explicit and handles both cases. What do you think?

Reply:
From your comment about using `set()` I just realized that it may be possible to add a test to cover the case where some qubits are repeated (purposefully for the test) and then ensure it gives the same result as when removing the duplicates. If needed, I can add this test.

Reply:
```suggestion
        highlighted_qubits = frozenset(kwargs.get("highlighted_qubits") or [])
```

Reply:
The reason I suggested set is because the styling code below uses a plenty of `qubit[0] in highlighted_qubits` expressions which are are fast for a set but require iteration over possibly all elements for lists or np.arrays.
(To be pedantic, we should use `frozenset` because there is currently no need to update highlighted_qubits once resolved from an argument.)  The issues with None can be avoided by using an empty tuple instead of None as the default value of highlighted_qubits.  Also, the documented type of the `highlighted_qubits` argument should be `Iterable[GridQubit]` instead of the `list` because that would include lists, tuples, sets, and numpy arrays that are currently used in testing.

To resolve `highlighted_qubits` from kwargs with a definite frozenset type, one could use
`highlighted_qubits = frozenset(kwargs.get("highlighted_qubits", ()))`.  After that the `if highlighted_qubits` expression should work without errors.

Reply:
Thanks for your explanation. But if `highlighted_qubits` is a numpy array, `frozenset(kwargs.get("highlighted_qubits") or [])` still gives those errors. If we use this, then we should enforce `highlighted_qubits` is necessarily a list.

Reply:
if you are worried about numpy then do this
```suggestion
        highlighted_qubits = _as_set(kwargs.get("highlighted_qubits", ()))
```

and define `_as_set` as 
```py3
def _as_set(qubits: Iterable['cirq.Qid']) -> FrozenSet['cirq.Qid']:
    if isinstance(qubits, np.array):
       return frozenset(qubits.reshape((-1,)).tolist())
    return frozenset(qubits)
```

Reply:
Thanks, this works now!